pkg/apis/postgres-operator.crunchydata.com/v1beta1/crunchy_bridgecluster_types.go

```diff
@@ -79,70 +87,92 @@ type CrunchyBridgeClusterSpec struct {
+	Name string `json:"name"`
+
+	// The name of the Secret that will hold the role credentials.
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	// +kubebuilder:validation:Type=string
+	SecretName string `json:"secretName"`
+}
+
 // CrunchyBridgeClusterStatus defines the observed state of CrunchyBridgeCluster
 type CrunchyBridgeClusterStatus struct {
-	// The ID of the postgrescluster in Bridge, provided by Bridge API and null until then.
+	// The name of the cluster in Bridge.
+	// +optional
+	ClusterName string `json:"name,omitempty"`
+
+	// conditions represent the observations of postgres cluster's current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors={"urn:alm:descriptor:io.kubernetes.conditions"}
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// The Hostname of the postgres cluster in Bridge, provided by Bridge API and null until then.
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// The ID of the postgres cluster in Bridge, provided by Bridge API and null until then.
 	// +optional
 	ID string `json:"id,omitempty"`
 
+	// Whether the cluster is high availability, meaning that it has a secondary it can fail
+	// over to quickly in case the primary becomes unavailable.
+	// +optional
+	IsHA bool `json:"isHa"`
+
+	// Whether the cluster is protected. Protected clusters can't be destroyed until
+	// their protected flag is removed
+	// +optional
+	IsProtected bool `json:"isProtected,omitempty"`
+
+	// The cluster's major Postgres version.
+	// +optional
+	MajorVersion int `json:"majorVersion"`
+
 	// observedGeneration represents the .metadata.generation on which the status was based.
 	// +optional
 	// +kubebuilder:validation:Minimum=0
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 
-	// conditions represent the observations of postgrescluster's current state.
+	// The cluster upgrade as represented by Bridge
 	// +optional
-	// +listType=map
-	// +listMapKey=type
-	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors={"urn:alm:descriptor:io.kubernetes.conditions"}
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	OngoingUpgrade []*UpgradeOperation `json:"ongoingUpgrade,omitempty"`
 
-	// The cluster as represented by Bridge
+	// The ID of the cluster's plan. Determines instance, CPU, and memory.
 	// +optional
-	Cluster *ClusterStatus `json:"clusterStatus,omitempty"`
+	Plan string `json:"planId"`
 
-	// The cluster upgrade as represented by Bridge
+	// Most recent, raw responses from Bridge API
 	// +optional
-	ClusterUpgrade *ClusterUpgrade `json:"clusterUpgradeResponse,omitempty"`
-}
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:validation:Type=object
+	Responses APIResponses `json:"responses"`
```

Comment:
Ah, ok, so we use this to shove the raw response from cluster, status, upgrade?

Reply:
Yep.